packages/enzyme-test-suite/test/ShallowWrapper-spec.jsx

```diff
@@ -2122,34 +2122,20 @@ describe('shallow', () => {
           ]);
         });
 
-        describeIf(is('0.13 || 15 || > 16'), 'setContext', () => {
+        describeIf(is('0.13 || 15 || >= 16'), 'setContext', () => {
           it('calls expected methods when receiving new context', () => {
             const wrapper = shallow(<Foo />, options);
             expect(spy.args).to.deep.equal([
               ['componentWillMount'],
               ['render'],
             ]);
             spy.resetHistory();
-            wrapper.setContext({ foo: 'foo' });
-            expect(spy.args).to.deep.equal([
-              ['componentWillReceiveProps'],
-              ['shouldComponentUpdate'],
-              ['componentWillUpdate'],
-              ['render'],
-            ]);
-          });
-        });
 
-        describeIf(is('16'), 'setContext', () => {
-          it('calls expected methods when receiving new context', () => {
-            const wrapper = shallow(<Foo />, options);
-            expect(spy.args).to.deep.equal([
-              ['componentWillMount'],
-              ['render'],
-            ]);
-            spy.resetHistory();
             wrapper.setContext({ foo: 'foo' });
+
             expect(spy.args).to.deep.equal([
+              // TODO: figure out what this test should actually be
+              ...(is('< 16 || ~16.3') ? [['componentWillReceiveProps']] : []),
```

Comment:
even with this, it's failing in 16.3 with the 16 adapter - but if i remove the `~16.3` from here, it fails with the 16.3 adapter.

@mul53 could you try to make more sandboxes with react 13, 14, 15, 16.2, 16.3, and 16.4, to see what this test should actually be doing?

Reply:
okay i will do that, should have them ready by tomorrow